src/vegasflow/vflow.py

```diff
@@ -352,6 +352,54 @@ def _run_iteration(self):
         return res, sigma
 
 
+class DistributedVegasFlow(VegasFlow):
+
+    def __init__(self, n_dim, n_events, **kwargs):
+        if "list_devices" in kwargs:
+            logger.warning("DistributedVegasFlow does not accept an input list of devices")
+            kwargs.pop("list_devices")
+        super().__init__(n_dim, n_events, list_devices = None, **kwargs)
+
+    def run_event(self, **kwargs):
+        """ Overrides ``MonteCarlo.run_event`` to distribute
+        the workload using dask
+        """
+        if not self.event:
+            raise RuntimeError("Compile must be ran before running any iterations")
+        # Run until there are no events left to do
+        events_left = self.n_events
+        events_to_do = []
+        percentages = []
+        # Fill the array of event distribution
+        # If using multiple devices, decide the policy for job sharing
+        pc = 0.0
+        while events_left > 0:
+            ncalls = min(events_left, self.events_per_run)
+            pc += ncalls / self.n_events * 100
+            percentages.append(pc)
+            events_to_do.append(ncalls)
+            events_left -= self.events_per_run
+
+        # Enable the slurm cluster
+        from dask_jobqueue import SLURMCluster # pylint: disable=import-error
```

Comment:
Right, but in principle we can delegate the client/cluster allocation to the user, right? (pass as variable)

Reply:
Yes for the cluster. I was thinking on adding a `setup_distribute` method that takes a cluster as argument.

The client instead I think it needs to be done by veasflow/`monte_carlo`. Dask cannot pickle itself so if the user allocates the client then the user needs to do the accumulation.